index.js

```diff
@@ -88,7 +88,11 @@ function processDependencies(_dependencies) {
   return dependencies.reduce((acc, component) => {
     // Add the main fragment if one is provided
     if (component.fragment) {
-      acc.push(component.fragment)
+      acc.push(
+        typeof component.fragment === 'string'
+          ? component.fragment
+          : print(component.fragment)
```

Comment:
[question] The `print` usage is to ensure that we're only dealing with strings here, right?

Reply:
Yeah, we're interpolating the return value in the string passed to `graphql-request`, so we have to convert DocumentNode fragments to strings to keep that working.

A future improvement could pass a DocumentNode to `graphql-request` instead. I skipped that for the time being since I don't know the specific AST manipulations I'd need to do to add fragments to a DocumentNode query.